src/algorithms/tracking/TrackPropagation.cc

```diff
@@ -290,8 +290,9 @@ namespace eicrecon {
         m_log->trace("    loc err = {:.4f}", static_cast<float>(covariance(Acts::eBoundLoc1, Acts::eBoundLoc1)));
         m_log->trace("    loc err = {:.4f}", static_cast<float>(covariance(Acts::eBoundLoc0, Acts::eBoundLoc1)));
 
+        int32_t surfaceID = 0; // default value...will be set in TrackPropagation factory
```

Comment:
I'm not sure this adds anything meaningful.  In the current implementation, `targetSurf` has just been created in the track propagation factory (and has not had a geometry ID set).  Even if a geometry ID is automatically assigned, it's not useful for this application so there's really nothing to be gained by (temporarily) setting the value this way.

Reply:
Your factory is not the only user of this algorithm, projectee DRICH surfaces should have the ID assigned. Your factory can be then also changed to assign meaningful ACTS geometry object IDs instead of overwriting them in each EDM object.